compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -360,7 +360,9 @@ fn check_gat_where_clauses(tcx: TyCtxt<'_>, associated_items: &[hir::TraitItemRe
                             tcx,
                             param_env,
                             item_def_id,
-                            tcx.explicit_item_bounds(item_def_id).to_vec(),
+                            tcx.explicit_item_bounds(item_def_id)
+                                .subst_identity_iter_copied()
+                                .collect::<Vec<_>>(),
```

Comment:
is the turbofish necessary here? Alternatively, you could do something like `subst_identity().to_vec()`, right?

Reply:
This maybe ties into a larger question (sorry for the long comment below):

Right now, `EarlyBinder<T>::subst_identity` has the bound `T: TypeFoldable`, so it's not actually available here. I was considering getting rid of the `TypeFoldable` bound but wasn't sure how important it is to how `EarlyBinder` should work -- i.e., right now you can only call `subst_identity` on things that you could also call `subst` on. (Maybe this property would come in handy in the future if doing larger refactorings?)

The lack of `subst_identity` is what motivated me to add `subst_identity_iter_copied` in this PR, as an identity version of `subst_iter_copied`.

Haha but now I'm worried that I'm overthinking and over-complicating this. If it's alright to do `subst_identity` on any inner type, then we don't have to add new methods on `EarlyBinder` in this PR, and every `subst_identity_iter_copied` would just become `subst_identity().iter()` (or in the case above, `subst_identity().to_vec()` as you mentioned). What do you think?

Reply:
Hmmmmmmm.. maybe we could implement `subst_identity` on anything that implements `TypeVisitable`? We probably don't need it to be `TypeFoldable`, but I could potentially see us using some folder to make sure that it's actually valid to subst identity?